### DISCUSSION TITLE
when i try to download this 720 video it always downloads 1080

### DISCUSSION THREAD

**Discussion by trup** [aclap-dev/video-downloadhelper#1157] [Help and Q&A]
### Your question

`https://www.ntv.ru/peredacha/SvoyaPravda/m78960/o772900`

### Addon & CoApp version and details

version: 9.1.0.42
target: mozilla
channel: stable
lang: ru
coapp: "checking"
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:129.0) Gecko/20100101 Firefox/129.0
{
    "used_history_button": true,
    "download_directory": "C:\\Users\\AnnA\\Videos",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\AnnA\\Videos\\новости"
        }
    },
    "concurrent_downloads_max": 5,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": true
    },
    "show_context_menu": false,
    "yt_warning": false,
    "use_wide_ui": false,
    "use_legacy_ui": false,
    "use_sidebar": false,
    "open_count_store": 16,
    "successfull_dl": 6,
    "never_show_successfull_dl_message": true,
    "record_download_history": true
}

**Comment by Mattmjs** [comment]
It does look like a detection issue but also the audio is missing, with the video being a dash stream. There is an alternative hls version
found in the image. A short test download of this got the 720p version with audio.

![Screenshot 2024-08-18 155938](https://github.com/user-attachments/assets/027d42ea-783a-445f-80fe-b3b48fb8ae09)

**Reply by GrampaWildWilly** [reply]
This is a curious case.  Further down in the Network Monitor there is an HLS master manifest.

![#01](https://github.com/user-attachments/assets/b3ec4073-2bad-45e3-bf14-3ecda22e257e)

However, when I ran ffprobe on it, I got 403 Forbidden.  On the other hand, the HLS master manifest you found via the json gives a perfectly accessible video:

[Russian Video mp4 Log.txt](https://github.com/user-attachments/files/16647534/Russian.Video.mp4.Log.txt)

Great download speed, too.  Looks like the site doesn't throttle your downloads.  The only slightly weird thing about this is that there is only the one audio track.  It's shared among the 4 video tracks.  But something like that should not bother VDH these days.

I think the problem here is that the standard way of doing things gets 403 Forbidden.  It takes a non-standard way of doing things to get it to work properly.

I did look at this with VDH.

![#02](https://github.com/user-attachments/assets/72de985b-cbcc-4c8d-bc09-9e35c72feb90)

The 720p variant is only 24 seconds long.  But when I downloaded the full video using the second entry on that VDH menu, I did not get the 1080p version of the show.  I got it at 256x144, and it did have audio.  But you'll note that the second entry is marked MPD.  In other words, it's a DASH stream.  Somehow, VDH fell back to using the DASH manifest instead of the HLS manifest.  I guess VDH got the 403 Forbidden error itself when it went looking for an HLS manifest so then it went looking for a DASH manifest.  But even that does not appear here in a standard way.  There is no entry in the Network Monitor with the type of DASH.  It's of type octet-stream, the entry I've highlighted in my image above with a box.  When I ran ffprobe on that, it gave essentially the equivalent of the HLS manifest that I used for my ffmpeg download above.  But the first video track listed in the DASH manifest was the 144p video.  Somehow, VDH is not properly recognizing the video resolutions in the DASH manifest.  It does seem to be recognizing them; if you drop down the list of resolutions in the VDH menu, it does list the 4 available resolutions in descending order.  But when it comes time to do the actual download, for some reason, VDH is not picking the resolution you're specifying in the VDH menu.

This one is very weird.

**Reply by GrampaWildWilly** [reply]
One other thing.  Our original poster does not appear to have the CoApp installed.  I would recommend you install that.

**Reply by trup** [reply]
I don't know why the coapp version wasn't included in the report, but it is installed.

version: 9.1.0.42
target: mozilla
channel: stable
lang: ru
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:129.0) Gecko/20100101 Firefox/129.0
{
    "used_history_button": true,
    "download_directory": "C:\\Users\\AnnA\\Videos",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\AnnA\\Videos\\новости"
        }
    },
    "concurrent_downloads_max": 5,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": true
    },
    "show_context_menu": false,
    "yt_warning": false,
    "use_wide_ui": false,
    "use_legacy_ui": false,
    "use_sidebar": false,
    "open_count_store": 28,
    "successfull_dl": 8,
    "never_show_successfull_dl_message": true,
    "record_download_history": true
}

**Reply by GrampaWildWilly** [reply]
That's good.  The only other thing I might suggest is to install the VDH beta.  But I'm running the beta & I didn't get the resolution I requested when I told VDH to download this, so I hesitate on that advice.  I think we need @paulrouget Paul to look at this.
